src/bootstrap/builder.rs

```diff
@@ -1543,6 +1543,9 @@ impl<'a> Builder<'a> {
             }
         }
 
+        rustdocflags.arg("-Zunstable-options");
+        rustdocflags.arg("--generate-link-to-definition");
```

Comment:
Hmm, is this the right place to put the check? I would expect it to go in `src/bootstrap/doc.rs` like for `--show-type-layout`. At least, this seems like it should be wrapped in an `if mode == Mode::Rustdoc`.

Reply:
@camelid --show-type-layout is only enabled for the compiler and tools, so it can't be enabled in builder unconditionally. I think it makes sense to use builder here so it doesn't have to be readded in each `Step` in doc.rs.

Good catch on `mode == Rustdoc` though :)

Reply:
We probably want this for all docs, right? In that case I don't believe there's a need to limit to just Mode::Rustdoc.

Reply:
> I think it makes sense to use builder here so it doesn't have to be readded in each Step in doc.rs.

Ok, I was wondering if that was the reason 👍

Reply:
@Mark-Simulacrum if we don't check the mode, it will be passed unconditionally for all steps, even compiles, right? I guess it doesn't really matter since most of the time rustdoc isn't invoked though.

Reply:
Right, RUSTDOCFLAGS should be fine to set unconditionally.